Sources/ImageOptions.swift

```diff
@@ -0,0 +1,335 @@
+//
+//  Image.swift
+//  Contentful
+//
+//  Created by JP Wright on 24.05.17.
+//  Copyright © 2017 Contentful GmbH. All rights reserved.
+//
+
+import Foundation
+import CoreGraphics
+
+extension Asset {
+
+    /**
+     The URL for the underlying media file with additional options for server side manipulations
+     such as format changes, resizing, cropping, and focusing on different areas including on faces,
+     among others.
+
+     - Parameter imageOptions: An array of `ImageOption` that will be used for server side manipulations.
+     - Throws: Will throw SDKError if the SDK is unable to generate a valid URL with the desired ImageOptions.
+     */
+    public func url(with imageOptions: [ImageOption] = []) throws -> URL {
+
+        // Check that there are no two image options that specifiy the same query parameter.
+        // https://stackoverflow.com/a/27624476/4068264z
+        // A Set is a collection of unique elements, so constructing them will invoke the Equatable implementation
+        // and unique'ify the elements in the array.
+        let uniqueImageOptions = Array(Set<ImageOption>(imageOptions))
+        guard uniqueImageOptions.count == imageOptions.count else {
+            throw SDKError.invalidImageParameters("Cannot specify two instances of ImageOption of the same case."
```

Comment:
So, actually you can specify two options at the same time: ```http://images.contentful.com/arqlnkt58eul/2ReMHJhXoAcy4AyamgsgwQ/788a7c2210c923644b4bc20deb9a1f78/lewis-carroll-1.jpg?w=100&w=2000``` will return an image with a width of 100.